### PR TITLE
aravis: add package_type

### PR DESCRIPTION
**Pull request by valgur**
TODO

- [x] `../src/src/meson.build:218:0: ERROR: Program 'glib-mkenums mkenums' not found or not executable`

### CODE REVIEW

**Review by ericLemanissier — APPROVED** [review]

**Comment by valgur** [comment]
Closing temporarily to avoid unnecessary load on the CI. Will reopen when I'm actively working on the PR again.

File: recipes/aravis/all/conanfile.py [hunk 1]
```diff
@@ -63,32 +65,33 @@ def layout(self):
...
+        if is_apple_os(self) and self.dependencies["glib"].options.shared:
             raise ConanInvalidConfiguration(
                 "macOS builds are disabled when glib is shared until "
                 "conan-io/conan#7324 gets merged to fix macOS SIP issue #8443"
             )
 
     def build_requirements(self):
-        self.tool_requires("meson/1.0.0")
+        self.tool_requires("meson/1.2.1")
         if hasattr(self, "settings_build") and cross_building(self):
```
Comment (ericLemanissier) [inline recipes/aravis/all/conanfile.py:91]:
```suggestion
```

File: recipes/aravis/all/conanfile.py [hunk 2]
```diff
@@ -63,32 +65,33 @@ def layout(self):
...
-            self.tool_requires("glib/2.75.2")
+            self.tool_requires("glib/<host_version>")
```
Comment (ericLemanissier) [inline recipes/aravis/all/conanfile.py:92]:
```suggestion
        self.tool_requires("glib/<host_version>")
```

**Review by ericLemanissier — COMMENTED** [review]
these two changes will fix conan 2 build (cf https://github.com/conan-io/conan-center-index/pull/18180)

**Review by ericLemanissier — APPROVED** [review]

**Review by uilianries — APPROVED** [review]
LGTM

**Pull request merged**